java/code/src/com/redhat/rhn/domain/server/MinionServer.java

```diff
@@ -326,10 +320,14 @@ private boolean updateServerPaths(Optional<Server> proxy, Optional<String> hostn
     }
 
     public Boolean isRebootNeeded() {
-        return rebootNeeded;
+        return getLastBoot() != null && rebootRequiredAfter != null && getLastBootAsDate().before(rebootRequiredAfter);
+    }
```

Comment:
I would use the `boolean` primitive as return type of the method now, since now it can never be null.